common/inc/c_pal/containing_record.h

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef CONTAINING_RECORD_H
+#define CONTAINING_RECORD_H
+
+#ifdef __cplusplus
+#include <cstddef>
+#include <cstdint>
+#else
+#include <stddef.h>
+#include <stdint.h>
+#endif
+
+//
+// Calculate the address of the base of the structure given its type, and an
+// address of a field within the structure.
+//
+
+#ifdef WIN32
+/*windows.h brings CONTAINING_RECORD*/
+#include "windows.h"
+#else
+#define CONTAINING_RECORD(address, type, field) ((type *)((uintptr_t)(address) - offsetof(type,field)))
+#endif
+
+#endif /* CONTAINING_RECORD_H */
```

Comment:
ha I guess this could have already been in the PAL with separate headers in the win32 and linux folders. Maybe it should just be two headers that way instead /consider